rustls/src/client/handy.rs

```diff
@@ -241,6 +241,44 @@ impl client::ResolvesClientCert for AlwaysResolvesClientCert {
     }
 }
 
+#[derive(Debug)]
+pub(super) struct AlwaysResolvesClientRawPublicKeys(Arc<sign::CertifiedKey>);
+
+impl AlwaysResolvesClientRawPublicKeys {
+    pub(super) fn new(private_key: Arc<dyn sign::SigningKey>) -> Result<Self, Error> {
```

Comment:
What's the motivation to not just take an `Arc<sign::CertifiedKey>` directly?

Reply:
Because it's in line with the `AlwaysResolvesClientCert::new()` function, where you also pass an `Arc<dyn sign::SigningKey>` and in the function an Arc<sign::CertifiedKey> is created. 

If it's better to just take `Arc<sign::CertifiedKey>` I can of course change it.

Reply:
I think it is better in this case to take `sign::CertifiedKey`, as it does precisely what 99.9% people want (the public key from their private key); without the potential for getting them from divergent sources. That does mean someone who has a private key impl that can't get the public key will need to provide a whole trait impl, I think that is uncommon.